applications/manufacturing/groovyScripts/routing/RoutingServices.groovy

```diff
@@ -0,0 +1,113 @@
+    if (parameters.workEffortId) {
+
+        lookupRouting.workEffortId = parameters.workEffortId
+        routingGS = from("WorkEffortGoodStandard").where(lookupRouting).filterByDate(filterDate).queryFirst()
+
+        // If the routing is not associated with our product and it's a variant, then check to see if it's virtual product has the routing
+        if (!routingGS) {
+            GenericValue virtualProductAssoc = from("ProductAssoc").where(productIdTo: parameters.productId, productAssocTypeId: "PRODUCT_VARIANT").filterByDate().queryFirst()
+            if (virtualProductAssoc) {
+                lookupRouting.productId = virtualProductAssoc.productId
+                // Consider the validity against a date passed as (optional) parameter
+                routingGS = from("WorkEffortGoodStandard").where(lookupRouting).filterByDate(filterDate).queryFirst()
+            }
+        }
+    }
+    // No workEffortId has been passed, so retrieve the first routing found for this product
+    else {
+        // Consider the validity against a date passed as (optional) parameter
+        // TODO: we should consider the quantity to select the best routing
+        routingGS = from("WorkEffortGoodStandard").where(lookupRouting).filterByDate(filterDate).queryFirst()
+        // If there are no routings associated with our product and it's a variant, then check to see if it's virtual product has a routing
+        if (!routingGS) {
+            GenericValue virtualProductAssoc = from("ProductAssoc").where(productIdTo: parameters.productId, productAssocTypeId: "PRODUCT_VARIANT").filterByDate(filterDate).queryFirst()
+            if (virtualProductAssoc) {
+                lookupRouting.productId = virtualProductAssoc.productId
+                lookupRouting.workEffortGoodStdTypeId = "ROU_PROD_TEMPLATE"
+                // Consider the validity against a date passed as (optional) parameter
+                // TODO: we should consider the quantity to select the best routing
+                routingGS = from("WorkEffortGoodStandard").where(lookupRouting).filterByDate(filterDate).queryFirst()
+            }
+        }
+    }
+    if (routingGS) {
+        lookupRouting.clear()
+        lookupRouting.workEffortId = routingGS.workEffortId
+        routing = from("WorkEffort").where(lookupRouting).queryOne()
+    }
+    else {
+        // The default routing is used when no explicit routing is associated to the product and the ignoreDefaultRouting is not equals to Y
+        if (!parameters.ignoreDefaultRouting || parameters.ignoreDefaultRouting.equals("N")) {
+            lookupRouting.clear()
+            lookupRouting.workEffortId = "DEFAULT_ROUTING"
+            routing = from("WorkEffort").where(lookupRouting).queryOne()
+        }
+    }
+    if (routing) {
+        Map lookupTasks = [workEffortIdFrom: routing.workEffortId, workEffortAssocTypeId: "ROUTING_COMPONENT"]
+        List tasksOrder = ["sequenceNum"]
```

Comment:
Hi @SebastianEcomify ,
Perhaps we can skip this list and instead use orderBy("sequenceNum") on the next line